### ISSUE TITLE
Documentation related question

### ISSUE THREAD

**Issue by ks1990cn** [RedouxG/SGT#1]
Why this project needed? we have Xunit framework, exactly what solution it overcomes? Will I be able to create behavioural test, end to end test cases using this? Is it a way to automate my testing like ChatGPT helps us?

**Comment by RedouxG** [comment]
Short answer:
Godot sucks at integrating anything related to testing into its API/IDE [(for example)](https://github.com/godotengine/godot-proposals/issues/432).

Long answer:
Its fairly difficult to find anything that would simply provide the use case of running a test in Godot. Most frameworks have elaborate built in integrations with IDE like xunit/nunit/mstest and because of that its difficult to set those up in a way that would allow the developer to run Godot engine dependent tests. 

There are some alternatives sure, for example [gdUnit4](https://github.com/MikeSchulze/gdUnit4), but it has a disclaimer "- CSharp support is currently disabled and will be reimplement with V4.2.0!", which is fairly often seen when it comes to anything regarding C# testing in Godot.

The small simple library I wrote does exactly what the average dev wants (at least from my perspective), which is:
1) Download x plugin that allows you to write tests that can be ran from godot editor level
2) Write a test with as little setup as possible
3) See if your godot derived class works and behaves as you would expect

Basically, as a lazy dev I want to write a test for some class that slices chunks in my game, run it, and not have the slightest care in the world if some weird integration issue makes the tests not run with 2 pages A4 of a stacktrace error.
